spec/datagrid/filters_spec.rb

```diff
@@ -292,5 +292,33 @@ class TestGrid8728
       expect(non_admin_filters).to_not include(:id)
       expect(non_admin_filters).to include(:name)
     end
+
+    context 'with delegation to attribute' do
+      let(:role) { OpenStruct.new('admin?' => admin) }
+      let(:klass) do
+        test_report_class do
+          attr_accessor :role
+          delegate :admin?, to: :role
+
+          scope { Entry }
+
+          filter(:id, :integer, if: :admin?)
+        end
+      end
+
+      subject { klass.new(role: role).filters.map(&:name) }
```

Comment:
Yes, the test fails with

```
Failure/Error: delegate :admin?, to: :role

Module::DelegationError:
  #<Class:0x007fa0ce453000>#admin? delegated to role.admin?, but role is nil: #<#<Class:0x007fa0ce453000> order: nil, descending: nil, id: nil>
```

I'm sorry but I'm not sure I understand the default-value assignment part of your comment. The call to ```filters```, which is an instance method executes a the block, where ```enabled?``` is called on the filter. And that is the issue I think.